src/datasources/notifications/entities/notification-subscription.entity.ts

```diff
@@ -0,0 +1,65 @@
+import { NotificationDevice } from '@/datasources/notifications/entities/notification-devices.entity';
+import { type NotificationSubscriptionNotificationType } from '@/datasources/notifications/entities/notification-subscription-notification-type.entity';
+import {
+  Column,
+  Entity,
+  Unique,
+  ManyToOne,
+  PrimaryGeneratedColumn,
+  OneToMany,
+} from 'typeorm';
+
+@Entity('notification_subscriptions')
+@Unique([
+  'chain_id',
+  'safe_address',
+  'push_notification_device',
+  'signer_address',
+])
+export class NotificationSubscription {
+  @PrimaryGeneratedColumn()
+  id!: number;
+
+  @ManyToOne(() => NotificationDevice, (device) => device.id, {
+    onDelete: 'CASCADE',
+  })
+  push_notification_device!: NotificationDevice;
+
+  @Column({
+    type: 'varchar',
+    length: 255,
+  })
+  chain_id!: string;
+
+  @Column({
+    type: 'varchar',
+    length: 42,
+  })
+  safe_address!: string;
+
+  @Column({
+    type: 'varchar',
+    nullable: true,
+    length: 42,
+  })
+  signer_address!: string | null;
```

Comment:
I think we should use the refinement functionality to call viem's `getAddress`, ensuring that these addresses are checksummed. We can then set the type to `0x${string}` as well.

Reply:
This will be addressed in a separate PR focused on notifications.